enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MTLSAuthenticator.java

```diff
@@ -0,0 +1,190 @@
+            if (Utils.tracingEnabled()) {
+                tracer = Utils.getGlobalTracer();
+                mtlsApiAuthenticatorSpan = Utils.startSpan(TracingConstants.MTLS_API_AUTHENTICATOR_SPAN, tracer);
+                mtlsApiAuthenticatorSpanScope = mtlsApiAuthenticatorSpan.getSpan().makeCurrent();
+                Utils.setTag(mtlsApiAuthenticatorSpan, APIConstants.LOG_TRACE_ID,
+                        ThreadContext.get(APIConstants.LOG_TRACE_ID));
+            }
+
+            AuthenticationContext authenticationContext = new AuthenticationContext();
+            KeyStore trustStore = retrieveTrustStore(requestContext);
+            Map<String, String> mtlsCertificateTiers = retrieveMtlsCertificateTiers(requestContext);
+            X509Certificate clientCert;
+            boolean authenticated = false;
+            String clientCertificateTier;
+            String clientCertificateAlias;
+
+            try {
+                clientCert = getClientCertificate(requestContext);
+                clientCertificateAlias = MtlsUtils.getMatchedCertificateAliasFromTrustStore(clientCert, trustStore);
+                if (clientCertificateAlias == null) {
+                    log.debug(String.format("Provided client certificate in request: %s is not in the truststore of " +
+                            "the API: %s", requestContext.getRequestPath(), requestContext.getMatchedAPI().getName()));
+                    clientCert = null;
+                }
+                if (clientCert != null) {
+                    authenticated = true;
+                    clientCertificateTier = MtlsUtils.getTier(clientCertificateAlias, mtlsCertificateTiers);
+                    if (clientCertificateTier != null) {
+                        if (!clientCertificateTier.equals("")) {
+                            authenticationContext.setTier(clientCertificateTier);
+                        }
+                    }
+                    String subjectDN = clientCert.getSubjectDN().getName();
+                    authenticationContext.setUsername(subjectDN);
+                }
+            } catch (CertificateException e) {
+                throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                        APISecurityConstants.API_AUTH_GENERAL_ERROR, "Invalid client certificate");
+            }
+
+            String apiName = requestContext.getMatchedAPI().getName();
+            String apiVersion = requestContext.getMatchedAPI().getVersion();
+            String apiUUID = requestContext.getMatchedAPI().getUuid();
+
+            authenticationContext.setAuthenticated(authenticated);
+            authenticationContext.setApiName(apiName);
+            authenticationContext.setApiUUID(apiUUID);
+            authenticationContext.setApiVersion(apiVersion);
+
+            return authenticationContext;
+        } finally {
+            if (Utils.tracingEnabled()) {
+                mtlsApiAuthenticatorSpanScope.close();
```

Comment:
mtlsApiAuthenticatorSpanScope is there any chance this to be null? if so, this line will throw an NPE

Reply:
I'm not sure if there is any chance for that. But all the other authenticators are using thre same way. [jwt](https://github.com/wso2/product-microgateway/blob/76cbfe666803b1ec2c301ed04c8bc2fad20e9088/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L314), [oauth](https://github.com/wso2/product-microgateway/blob/76cbfe666803b1ec2c301ed04c8bc2fad20e9088/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/oauth/OAuthAuthenticator.java#L136), [internalapikey](https://github.com/wso2/product-microgateway/blob/76cbfe666803b1ec2c301ed04c8bc2fad20e9088/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/InternalAPIKeyAuthenticator.java#L276)